### PR TITLE
Feature/better json usage

### PR DESCRIPTION
**Pull request by ghost**
This is concerned more with this change

```
        /// <summary>
        /// Serializes a localized content <see cref="IDictionary{TKey,TValue}"/> to JSON.
        /// </summary>
        public string Serialize(IDictionary<string, string> content)
        {
            using (var sw = new StringWriter())
            using (var jw = new JsonTextWriter(sw))
            {
                jw.WriteStartObject();
                foreach (var languageCode in SupportedLanguages)
                {
                    if (content.TryGetValue(languageCode, out var value))
                    {
                        jw.WritePropertyName(languageCode);
                        jw.WriteValue(value);
                    }
                }

                jw.WriteEndObject();
                return sw.ToString();
            }
        }
```

and this

```
        /// <summary>
        /// Deserializes JSON localized content to <see cref="IDictionary{TKey,TValue}"/>.
        /// </summary>
        public IDictionary<string, string> Deserialize(string json)
        {
            var item = JsonConvert.DeserializeObject<IDictionary<string, string>>(json);

            foreach (var key in item.Keys)
            {
                if (!SupportedLanguages.Contains(key))
                {
                    item.Remove(key);
                }
            }

            return item;
        }
```

Practically eliminating the usage of `JObject`, further more this opens the option of having a `SerializeAsync` since `JsonTextWriter` supports write async methods.

While some part also modified the localize property to add initial support for `Shallow` and `OneLevel` localization. `Deep` is not complete yet. That being said, all unit tests are passing and it's safe to merge and use this. `Deep` is treat as `OneLevel` at the moment.

I will probably add unit tests to the Depths once `Deep` is implemented.

### CODE REVIEW

File: samples/Xaki.Sample/Models/Models.cs [hunk 1]
```diff
@@ -27,13 +27,13 @@ public static void Initialize(DataContext context)
...
+                new Planet { Name = "{'en':'Earth','zh':'地球','ar':'الأرض','es':'Tierra','hi':'पृथ्वी','pt':'Terra','ru':'Земля́','ja':'地球','de':'Erde','el':'Γη'}" },
+                new Planet { Name = "{'en':'Mars','zh':'火星','ar':'المريخ','es':'Marte','hi':'मंगल','pt':'Marte','ru':'Марс','ja':'火星','de':'Mars','el':'Άρης'}" },
+                new Planet { Name = "{'en':'Jupiter','zh':'木星','ar':'المشتري','es':'Júpiter','hi':'बृहस्पति','pt':'Júpiter','ru':'Юпи́тер','ja':'木星','de':'Jupiter','el':'Δίας'}" },
+                new Planet { Name = "{'en':'Saturn','zh':'土星','ar':'زحل','es':'Saturno','hi':'शनि','pt':'Saturno','ru':'Сату́рн','ja':'土星','de':'Saturn','el':'Κρόνος'}" },
+                new Planet { Name = "{'en':'Uranus','zh':'天王星','ar':'أورانوس','es':'Urano','hi':'अरुण','pt':'Urano','ru':'Ура́н','ja':'天王星','de':'Uranus','el':'Ουρανός'}" },
+                new Planet { Name = "{'en':'Neptune','zh':'海王星','ar':'نبتون','es':'Neptuno','hi':'वरुण','pt':'Neptuno','ru':'Непту́н','ja':'海王星','de':'Neptun','el':'Ποσειδώνας'}" }
```
Comment (mehalick) [inline samples/Xaki.Sample/Models/Models.cs:36]: Wow not sure how these even worked before

File: src/Xaki.AspNetCore/TagHelpers/LocalizedEditor.cs [hunk 2]
```diff
@@ -129,22 +129,22 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
                         (function () {{
                             var getMaxWidth = function(elements) {{
-                                var width = 0;
```
Comment (mehalick) [inline src/Xaki.AspNetCore/TagHelpers/LocalizedEditor.cs:132]: Yeah the only reason I use `var` over `let` is for older browser support (I guess IE 10), will need to decide if I care about IE 10 but maybe it's time to move on.

File: tests/Xaki.Tests/ObjectLocalizerFacts.cs [hunk 3]
```diff
@@ -19,15 +19,26 @@ public abstract class TestBase
             };
         }
 
+        public class GetEmptyJsonString : TestBase
```
Comment (mehalick) [inline tests/Xaki.Tests/ObjectLocalizerFacts.cs:22]: Code coverage FTW!

File: tests/Xaki.Tests/ObjectLocalizerFacts.cs [hunk 4]
```diff
@@ -19,15 +19,26 @@ public abstract class TestBase
...
         {
             [Fact]
             public void ReturnsCorrectJsonForValidContent()
             {
                 var contents = new Dictionary<string, string>
                 {
-                    {Constants.LanguageCode1, Constants.AnyString1 },
-                    {Constants.LanguageCode2, Constants.AnyString2 }
+                    [Constants.LanguageCode1] = Constants.AnyString1,
+                    [Constants.LanguageCode2] = Constants.AnyString2
```
Comment (mehalick) [inline tests/Xaki.Tests/ObjectLocalizerFacts.cs:41]: I do like the C#6 dictionary initializer pattern better, need to get it into muscle memory

**Review by mehalick — APPROVED** [review]

File: samples/Xaki.Sample/Models/Models.cs (hunk 1 above)
Comment (ghost) [reply samples/Xaki.Sample/Models/Models.cs:36]: I think you added those new planets recently.

File: src/Xaki.AspNetCore/TagHelpers/LocalizedEditor.cs (hunk 2 above)
Comment (ghost) [reply src/Xaki.AspNetCore/TagHelpers/LocalizedEditor.cs:132]: `Let` IE go.

**Comment by ghost** [comment]
I never really got why we got an error when merging. The file in question `IObjectLocalizer` looked fine to me.

**Comment by mehalick** [comment]
Yeah no issues with the PR but the build uses a merge which was duplicating the method declarations, I did a quick post merge commit to remove the dupes.

File: samples/Xaki.Sample/Models/Models.cs (hunk 1 above)
Comment (mehalick) [reply samples/Xaki.Sample/Models/Models.cs:36]: I also left the original seed scripts in the initial commit db migration so these never actually run, am cleaning up now...

**Pull request merged**